src/Compilers/CSharp/Portable/Symbols/MergedNamespaceSymbol.cs

```diff
@@ -82,14 +83,30 @@ internal sealed class MergedNamespaceSymbol : NamespaceSymbol
 
             Debug.Assert(namespacesToMerge.Length != 0);
 
-            return (namespacesToMerge.Length == 1 && nameOpt == null)
-                ? namespacesToMerge[0]
-                : new MergedNamespaceSymbol(extent, containingNamespace, namespacesToMerge, nameOpt);
+            if (namespacesToMerge.Length == 1 && nameOpt is null)
+                return namespacesToMerge[0];
+
+            if (namespacesToMerge.Any(ns => ns is MergedNamespaceSymbol))
```

Comment:
> if (namespacesToMerge.Any(ns => ns is MergedNamespaceSymbol)) [](http://example.com/codeflow?start=11&length=62)

Does this really make a difference? It looks like we were already asserting that none of the ```namespacesToMerge``` are merged themselves.

Reply:
This code block is hit by the new test.

Can you point me to the previous assertion?

Reply:
> Can you point me to the previous assertion?

There is one at the end of the constructor.
```
#if DEBUG
            // We shouldn't merged namespaces that are already merged.
            foreach (NamespaceSymbol ns in namespacesToMerge)
            {
                Debug.Assert(ns.ConstituentNamespaces.Length == 1);
            }
#endif
```

Reply:
Oh, nice, let me see if the new test fails without my flattening changes.

Reply:
Yes, the new test fails without my change. I can remove the added assertion though.

Reply:
> Yes, the new test fails without my change. I can remove the added assertion though.

You mean the new assert fails, but the old one doesn't?

Reply:
Both asserts fail with the new test. The assertion previously added in this PR was redundant.

Reply:
> Both asserts fail with the new test. The assertion previously added in this PR was redundant.

Then I suggest opening an issue with the new unit test and leaving the code unchanged.

Reply:
➡️ Filed #67898

Reply:
> Then I suggest opening an issue with the new unit test and leaving the code unchanged.


Either I am missing something, or I stand by my opinion. I.e. the fix should be elsewhere. ```Create shouldn't be called with the input that is not flat.